added impsingle

lesyna opened the pull request:
My first pull request, adding impsingle

wlammen commented:
you need to limit the line width to at most 80 characters per line.

mazsa commented:
You should check the Contributing file esp. the 
["Formatting recommendation prior to submitting a pull request"](https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request) part. First, it is recommended to download and install the metamath.exe program, e.g. from the [landing page](https://us.metamath.org/) of the Metamath site, and check set.mm containing your theorem. Then you should commit your theorem to your github, and check your verifiers here: https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml . You should commit to the github of metamath only if they are *all* green.

lesyna commented:
I tried several variations, but my last attempt gives:

Error: Process completed with exit code 1.

It is interesting that Metamath.exe only gives this error after “git push”. 

> On Jul 18, 2023, at 11:37 AM, Wolf Lammen ***@***.***> wrote:
> 
> 
> you need to limit the line width to at most 80 characters per line.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1640749328>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NHDE5NF4XROFPHYU5LXQ3JWPANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

mazsa commented:
You should check the Contributing file esp. the 
["Formatting recommendation prior to submitting a pull request"](https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request) part. First, it is recommended to download and install the metamath.exe program, e.g. from the [landing page](https://us.metamath.org/) of the Metamath site, and check set.mm containing your theorem. Then you should commit your theorem to your github, and check your verifiers here: https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml . You should commit to the github of metamath only if they are *all* green.

lesyna commented:
With respect to the best of my knowledge I am doing exactly that!

My local Metamath.exe returns no error.

After commit to GitHub, all verifiers are green.

Only after push,  I get all green except Metamath.exe

Larry



> On Jul 18, 2023, at 2:17 PM, mazsa ***@***.***> wrote:
> 
> 
> You should check the Contributing file esp. the
> "Formatting recommendation prior to submitting a pull request" <https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request> part. First, it is recommended to download and install the metamath.exe program, e.g. from the landing page <https://us.metamath.org/> of the Metamath site, and check set.mm containing your theorem. Then you should commit your theorem to your github, and check your verifiers here: https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml <https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml> . You should commit to the github of metamath only if they are all green.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1641003465>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NGW5RMZYZJ63GHOHADXQ34NJANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

mazsa commented:
> After commit to GitHub, all verifiers are green. Only after push, I get all green except Metamath.exe

Hm... strange: they should be the same verifiers. OK, now your last one https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml is green indeed. Since there is a "This workflow requires approval from a maintainer" message above, you should wait for one of them to start verifiers. (And of course, green verifiers are only necessary not sufficient conditions: admins usually insist to open a personal sandbox called Mathbox, etc.)

digama0 commented:
The current revision is not likely to pass the metamath.exe check, because it is not rewrapped correctly. You can see in the output https://github.com/lesyna/set.mm/actions/runs/5592221367/jobs/10225280105 that it is recommending to change the indentation level of the comment, as well as the proof. You should run `scripts/rewrap set.mm` to format the proof.

lesyna commented:
OK thanks, I missed the rewrap requirement because it was not on this page:

https://github.com/metamath/set.mm/wiki/Getting-started-with-contributing <https://github.com/metamath/set.mm/wiki/Getting-started-with-contributing>

It was very confusing for a first time user to get the green light on the verifiers before “git push”, and then get the red light afterwards.

Anyway, after rewrap all went well. Thanks for your help!

Larry


> On Jul 18, 2023, at 3:15 PM, Mario Carneiro ***@***.***> wrote:
> 
> 
> The current revision is not likely to pass the metamath.exe check, because it is not rewrapped correctly. You can see in the output https://github.com/lesyna/set.mm/actions/runs/5592221367/jobs/10225280105 <https://github.com/lesyna/set.mm/actions/runs/5592221367/jobs/10225280105> that it is recommending to change the indentation level of the comment, as well as the proof. You should run scripts/rewrap set.mm to format the proof.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1641061928>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NE5TGPYPLTKT6MB73LXQ4DGXANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

digama0 commented:
It appears that the manual all-green gh actions run https://github.com/lesyna/set.mm/actions/runs/5592902058 was in fact testing https://github.com/lesyna/set.mm/commit/b30c9df0da05f236c4e73ff5af23517bf88fd246 (you can see this commit mentioned at https://github.com/lesyna/set.mm/actions/runs/5592902058/jobs/10225963398#step:2:412), which is not your PR, it is the unmodified master branch.

wlammen commented on set.mm:
```diff
@@ -12502,6 +12502,22 @@ the logical conjunction ( ~ df-an ).
+     ~ peirce , ~ ax-1 and ~ ax-2 } must be identical.  (Contributed by Larry
+     Lesyna and Jeffrey P. Machado, 18-Jul-2023.) $)
+  impsingle $p |- ( ( ( ph ->
+ps ) -> ch ) -> ( ( ch -> ph ) -> ( th -> ph ) ) ) $=
```
I suggest to break up the long expression
`( ( ( ph -> ps ) -> ch )`    ` -> ( ( ch -> ph ) -> ( th -> ph ) ) )`
at the point, where antecedents are separated from the consequent, for better readability and alignment with similar long expressions (see pm2.86 for example).  It further helps to indent the second portion such that it starts at a the same or a later position than the first one.  Only use space characters for indentation, no tabulators.

`From this axiom and ~ ax-mp , { ~ peirce , ~ ax-1 , ~ ax-2 } can be proved`

Do it.  To be accepted as (yet) another axiomatic foundation of propositional logic, the equivalence with the old system must be shown first.  In addition ax-3 should follow from peirce, ax-1, ax-2, by our traditional techniques.

The current content of the PR cannot be accepted, because of the mentioned deficiencies.  But even if they are resolved, there might be a discussion about whether it is important enough to be added in the main portion, and if so, where to add it.

You can instead create your own mathbox, where your idea can live independently of my or anyone else's opinion.

lesyna commented:
I am new to using GitHub and I was following the instructions, to the best of my ability, at:

https://github.com/metamath/set.mm/wiki/Getting-started-with-contributing <https://github.com/metamath/set.mm/wiki/Getting-started-with-contributing>

It was suggested that I do a verification on my commit before the PR using:

https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml <https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml>

It appears that I did something incorrectly at this point, but I am not sure what went wrong, because the wrong branch was verified.

I plan to make future PRs but would like to verify them correctly next time.

Could you please tell me what I should do to correctly verify my commit next time?

Larry



> On Jul 18, 2023, at 3:34 PM, Mario Carneiro ***@***.***> wrote:
> 
> 
> It appears that the manual all-green gh actions run https://github.com/lesyna/set.mm/actions/runs/5592902058 <https://github.com/lesyna/set.mm/actions/runs/5592902058> was in fact testing ***@***.*** <https://github.com/lesyna/set.mm/commit/b30c9df0da05f236c4e73ff5af23517bf88fd246> (you can see this commit mentioned at https://github.com/lesyna/set.mm/actions/runs/5592902058/jobs/10225963398#step:2:412 <https://github.com/lesyna/set.mm/actions/runs/5592902058/jobs/10225963398#step:2:412>), which is not your PR, it is the unmodified master branch.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1641076845>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NBSBX2AVCKYXP223ODXQ4FOLANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

icecream17 commented:
i think the correct file is https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md

(usually in simpler prs the only necessary section is https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request)

lesyna commented:
I did not find in either of these documents the procedure for doing a verify _before_ a PR.

Larry



> On Jul 19, 2023, at 2:21 PM, icecream17 ***@***.***> wrote:
> 
> 
> i think the correct file is https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md <https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md>
> (usually in simpler prs the only necessary section is https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request <https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request>)
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1642774435>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NDDG6ISEFF4Y547KNDXRBFUHANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

mazsa commented:
Plus, in addition to the green verification, you should read carefully the comment of @wlammen above, esp. his "Do it" paragraph. Please follow his advice, put those results in a personal Mathbox (to find examples, search for "Mathbox for" in set.mm), check the green lights, and open a PR:)

FYI: "Many contributors have or create a personal "mathbox" - a sandbox of work that is visible to others. Changes to someone's own mathbox still go through a review, but usually it's cursory - we generally just want to ensure that those changes don't interfere with or confuse others. Mathbox changes must still pass all verifiers, though, and that is a very rigorous check. Mathboxes should normally only be changed by the owner of the mathbox (unless it's a general improvement that applies across the database)."

mazsa commented:
> I did not find in either of these documents the procedure for doing a verify _before_ a PR. Larry

1. You can verify it offline before a PR this way:
https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request

2. You can verify it online, *still before a PR* this way:
"you can check in relative privacy, still in your
cloned repo, here
https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml
whether all your verifiers will pass = be green"
https://groups.google.com/g/metamath/c/hq9js-wHpb0/m/wphg0wZTBQAJ

icecream17 commented:
I think there isn't a need to verify by yourself, since when a pull request is submitted, verification is done automatically for you. you can see the results at https://github.com/metamath/set.mm/pull/3303/checks (it passes!)

(in particular verifying before submitting isn't necessary, only verifying before _merging_. example: https://github.com/metamath/set.mm/pull/3299 where my first commit fails, and the second fixes that)

(does lesyna have the ability to merge?? I don't)
![image](https://github.com/metamath/set.mm/assets/58114641/a6c62294-e8df-4178-a3a4-220033eeb589)

(sorry for any confusion. it was in this POV that I made the above comment, I thought something else was being talked about.)

----

huh, it's interesting that this is accepted. i've never seen any part of a theorem start on column 1 before
![image](https://github.com/metamath/set.mm/assets/58114641/c597ba48-2b6f-422a-b229-ef2fddbaec9f)

lesyna commented:
Thank you for your response.  There are two outstanding issues I am trying to resolve:

1) What I should submit next. I have read and understood Wolf Lammen’s comments. I am working on that!

2) How to do the submission. This is where I am asking for help. You say “check the green lights, and open a PR”. Please elaborate on how to do that. My previous attempts to do a verify before PR did give me green lights but failed to verify after the PR.

Larry



> On Jul 19, 2023, at 3:07 PM, mazsa ***@***.***> wrote:
> 
> 
> Plus, in addition to the green verification, you should read carefully the comment of @wlammen <https://github.com/wlammen> above, esp. his "Do it" paragraph. Please follow his advice, put those results in a personal Mathbox (to find examples, search for "Mathbox for" in set.mm), check the green lights, and open a PR:)
> 
> FYI: "Many contributors have or create a personal "mathbox" - a sandbox of work that is visible to others. Changes to someone's own mathbox still go through a review, but usually it's cursory - we generally just want to ensure that those changes don't interfere with or confuse others. Mathbox changes must still pass all verifiers, though, and that is a very rigorous check. Mathboxes should normally only be changed by the owner of the mathbox (unless it's a general improvement that applies across the database)."
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1642819950>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NH2MJABZQ6AH7OD4T3XRBLB5ANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

lesyna commented:
My previous email was written before I received this.

I did #2 below and all verifiers did pass with green, but my PR did not verify.

Larry



> On Jul 19, 2023, at 3:21 PM, mazsa ***@***.***> wrote:
> 
> 
> I did not find in either of these documents the procedure for doing a verify before a PR. Larry
> 
> You can verify it offline before a PR this way:
> https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request <https://github.com/metamath/set.mm/blob/develop/CONTRIBUTING.md#formatting-recommendation-prior-to-submitting-a-pull-request>
> You can verify it online, still before a PR this way:
> "you can check in relative privacy, still in your
> cloned repo, here
> https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml <https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml>
> whether all your verifiers will pass = be green"
> https://groups.google.com/g/metamath/c/hq9js-wHpb0/m/wphg0wZTBQAJ <https://groups.google.com/g/metamath/c/hq9js-wHpb0/m/wphg0wZTBQAJ>
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1642839896>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577ND24LZLR3QI3WYQQGDXRBMV7ANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

mazsa commented:
> 1) What I should submit next. I have read and understood Wolf Lammen’s comments. I am working on that! 

OK perfect:)

> 2) How to do the submission. This is where I am asking for help. You say “check the green lights, and open a PR”. Please elaborate on how to do that. My previous attempts to do a verify before PR did give me green lights but failed to verify after the PR. 

I think the explanation is what Mario wrote above: "It appears that the manual all-green gh actions run https://github.com/lesyna/set.mm/actions/runs/5592902058 was in fact testing https://github.com/lesyna/set.mm/commit/b30c9df0da05f236c4e73ff5af23517bf88fd246 (you can see this commit mentioned at https://github.com/lesyna/set.mm/actions/runs/5592902058/jobs/10225963398#step:2:412), which is not your PR, it is the unmodified master branch."

It seems that you run tests manually, and sometimes you are testing something which is not what you think you are testing. Try this simple workflow: 1. edit set.mm in an editor say VS Code, 2. save it there, 3. commit the changes to your local master from Github Desktop, and then 4. push it to your origin master from Github Desktop. This initiates verifications *automatically*, you can check the process here: https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml
These tests above and these tests below
https://github.com/metamath/set.mm/actions/workflows/verifiers.yml
are the *exact same* tests, so their results can not be other than the same: if you do not initiate manual tests but follow the workflow above, you can check before PR whether your lights will be green after PR.

jkingdon commented:
First of all, thank you for your persistence! As you've noticed, there are a bunch of fiddly bits to contributing and I don't think anyone expects you to understand the various kinds of checks the first time (certainly I don't). There is some documentation, but to be honest you are the expert (rather than experienced contributors) about what you were hoping to get from the documentation so just let me say that pull requests to the documentation, or perhaps more likely just a github issue or mailing list post describing what you were expecting to see where, are also accepted.

As for the substance of this contribution, I'd agree with the sentiment that a mathbox is the obvious place for this (and indeed a lot of contributions). Hopefully the word "mathbox" isn't too intimidating, it just means putting the proof later in the file under a "Mathbox for Larry" section header, as described at https://us.metamath.org/mpeuni/mathbox.html

lesyna commented:
Thanks for the recent posts giving specific details about how to do verification before the PR.

For the record, and in case this might help another first time user, here is what I did and my understanding of what went wrong.

I am not using GitDesktop at all.  I followed the instructions at:

https://github.com/metamath/set.mm/wiki/Getting-started-with-contributing <https://github.com/metamath/set.mm/wiki/Getting-started-with-contributing>

Which is wonderfully detailed about how to get GitHub going for a new users, like me. Although it does not appear on that page, I was told to use this link to verify my commit before PR:

https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml <https://github.com/lesyna/set.mm/actions/workflows/verifiers.yml>

Following this link, I was presented with a web page that offered a button “Run workflow”. I guessed that was the button to do the verification and it then presented a menu with a prominent green “Run workflow” button which I couldn’t resist pressing.  What I didn’t see at that time was that Branch: develop was the default, not my branch “proplogic”. So I was verifying the original set.mm and not my changes to set.mm as Mario and Peter were saying. Now that I understand that I have to select “proplogic” each time , all should be well for future verifications.

During this time I have been working on addressing the issues raised by Wolf Lammen about the content of the submission, and that is where I am directing my efforts.

Larry



> On Jul 20, 2023, at 6:29 AM, Jim Kingdon ***@***.***> wrote:
> 
> 
> First of all, thank you for your persistence! As you've noticed, there are a bunch of fiddly bits to contributing and I don't think anyone expects you to understand the various kinds of checks the first time (certainly I don't). There is some documentation, but to be honest you are the expert (rather than experienced contributors) about what you were hoping to get from the documentation so just let me say that pull requests to the documentation, or perhaps more likely just a github issue or mailing list post describing what you were expecting to see where, are also accepted.
> 
> As for the substance of this contribution, I'd agree with the sentiment that a mathbox is the obvious place for this (and indeed a lot of contributions). Hopefully the word "mathbox" isn't too intimidating, it just means putting the proof later in the file under a "Mathbox for Larry" section header, as described at https://us.metamath.org/mpeuni/mathbox.html <https://us.metamath.org/mpeuni/mathbox.html>
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1643929909>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NH6QS7SOCOKT327XIDXREXENANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

lesyna commented:
I added my mathbox at the end of set.mm and formatted it as best I could, similar to the others.  Metamath.exe verified set.mm correctly, and the HTML for the theorem in my mathbox rendered OK. So I tried to do the rewrap process. I got an error at the end…

MM> read set.mm
Reading source file "set.mm"... 43243324 bytes
43243324 bytes were read into the source buffer.
The source has 205866 statements; 2746 are $a and 40948 are $p.
No errors were found.  However, proofs were not checked.  Type VERIFY PROOF *
if you want to check them.
MM> write source set.mm /rewrap
The input file will be renamed set.mm~1.
Writing "set.mm"...
205866 source statement(s) were written.
MM> erase
Metamath has been reset to the starting state.
MM> read set.mm
Reading source file "set.mm"... 43243324 bytes
43243324 bytes were read into the source buffer.
The source has 205866 statements; 2746 are $a and 40948 are $p.
No errors were found.  However, proofs were not checked.  Type VERIFY PROOF *
if you want to check them.
MM> save proof */compressed/fast
Reformatting and saving (but not recompressing) all proofs...
Remember to use WRITE SOURCE to save changes permanently.
MM> verify markup */file_skip/top_date_skip
Checking statement label conventions...
Checking latexdef, htmldef, althtmldef...
Checking statement comments...
Checking section header comments...
Checking bibliographic references...
Abort trap: 6

I’d appreciate any advice about how to eliminate the “Abort trap:6”

Larry



> On Jul 20, 2023, at 6:29 AM, Jim Kingdon ***@***.***> wrote:
> 
> 
> First of all, thank you for your persistence! As you've noticed, there are a bunch of fiddly bits to contributing and I don't think anyone expects you to understand the various kinds of checks the first time (certainly I don't). There is some documentation, but to be honest you are the expert (rather than experienced contributors) about what you were hoping to get from the documentation so just let me say that pull requests to the documentation, or perhaps more likely just a github issue or mailing list post describing what you were expecting to see where, are also accepted.
> 
> As for the substance of this contribution, I'd agree with the sentiment that a mathbox is the obvious place for this (and indeed a lot of contributions). Hopefully the word "mathbox" isn't too intimidating, it just means putting the proof later in the file under a "Mathbox for Larry" section header, as described at https://us.metamath.org/mpeuni/mathbox.html <https://us.metamath.org/mpeuni/mathbox.html>
> —
> Reply to this email directly, view it on GitHub <https://github.com/metamath/set.mm/pull/3303#issuecomment-1643929909>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AF577NH6QS7SOCOKT327XIDXREXENANCNFSM6AAAAAA2OYSNQA>.
> You are receiving this because you authored the thread.
>

jkingdon commented:
> For the record, and in case this might help another first time user, here is what I did and my understanding of what went wrong

Ah, thanks for posting. Doesn't seem like there is any particular edit to the documentation which is called for and glad to hear you figured it out.

jkingdon commented:
> Abort trap: 6

You have the worst luck!

I'm not sure how confidently I can say this, but chances are it is a bug in metamath-exe and can be reported here: https://github.com/metamath/metamath-exe/issues . Especially if it always happens given the same input file.

If you just want to work around it, I guess experiment with moving your new theorem around a little? I mean, there might be some kind of syntax which triggers the bug, but if it is something like that, it really would be better if metamath-exe gave a real error message.

(There are, I suppose, more obscure possibilities, but given that you successfully got it to rewrap before you moved the theorem, it would appear to be related to the input file contents).

GinoGiotto commented:
> Abort trap: 6

This might be worth investigating. An exact copy of set.mm used by @lesyna is needed to check if the event is reproducible.

wlammen commented:
I added a mathbox to the end of set.mm and ran rewrap.sh, without failure.  So while it is certainly favorable to learn more about the abort and its causes, it should not impede the work here.  If necessary I add this mathbox template to the end of set.mm in my current pull request, for your convenience.

The addition looks like this (use empty lines as indicated here, do not touch lines before or after the ones shown here):

`$( (End of Kunhao Zheng's mathbox.) $)`
``
``
``
``
`$(`
`#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#`
`  Mathbox for XXX`
`#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#`
`$)`
``
``
``
``
`  ${`
`    xxx-embant.1 $e |- ph $.`
`    xxx-embant.2 $e |- ( ps -> ch ) $.`
`    $( A true wff can always be added as a nested antecedent to an antecedent.`
`       Note: this theorem is intuitionistically valid.  (Contributed by Wolf`
`       Lammen, 4-Oct-2013.)  (Proof modification is discouraged.)`
`       (New usage is discouraged.) $)`
`    xxx-embant $p |- ( ( ph -> ps ) -> ch ) $=`
`      ( wi imim2i mpi ) ABFACDBCAEGH $.`
`  $}`
``
``
`$( (End of XXX's mathbox.) $)`
``
``
`$( End $[ set-mbox.mm $] $)`

icecream17 commented:
according to https://stackoverflow.com/a/41340760 `Abort trap: 6` is one of the possible side-effects of Undefined Behavior, so this bug will probably be hard to find unless a file consistently causes the bug.

benjub commented:
You should write the reference in a more standard manner, for instance:
```
... This axiom is due to Lukasiewicz, see Lukasiewicz, _The Shortest Axiom of the
Implicational Calculus of Propositions,_ Proceedings of the Royal Irish Academy,
Section A, vol. 52 (1948-1950), 25--33.
```
Once you prove the equivalence over ax-mp of {impsingle} with {ax-1, ax-2, peirce}, you can move it as Subsection "1.3.2 Implicational calculus", right after Subsection "1.3.1 Minimal implicational calculus", and you can take its comment as a model.

This will indeed be a nice addition.

jkingdon commented:
> This will indeed be a nice addition.

I should say explicitly, it is fine to put this in a mathbox and the requirements there are much less demanding (mostly, that it pass the automated checks).

This can be a first step toward something more ambitious in the future.

lesyna commented:
I appreciate all the help.  I closed this pull request because it was simpler for me to just erase everything and reinstall Metamath from GitHub. I had been using the code downloaded at metamath.org. That seems to solve the abort 6 message and I submitted a new pull request #3310.

jkingdon commented:
> reinstall Metamath from GitHub. I had been using the code downloaded at metamath.org. That seems to solve the abort 6 message

Ah of course. Don't know why I didn't think to ask about the metamath version but of course it makes sense that upgrading might help.
